x-optparse/src/X/Options/Applicative.hs

```diff
@@ -86,6 +90,24 @@ dispatch p = getArgs >>= \x -> case x of
             >>= handleParseResult . removeError
   _  -> execParser (info (p <**> helper) idm)
 
+cli :: Show a => [Char] -> [Char] -> [[Char]] -> Parser a -> (a -> IO b) -> IO b
```

Comment:
newtypes? or at least a usage example for this function from strings and strings and strings to things